File: solutions/problem-003/solution.java
import java.util.Scanner;

public class Solution{
    public static void main(String[] args){
        Scanner in = new Scanner(System.in);
        System.out.printf("Enter i Value:  ");
        long n = in.nextLong();
        long number = n;
        long largestPrimeFactor = n;
        long i = 2;
        while (i <= n && n != 1) {
            if (n % i == 0) {
                n = n / i;
                largestPrimeFactor = i;
            }
            else {
                i = i+1;
            }
        }
        System.out.println("Largest prime factor of "+ number + " is : "+ largestPrimeFactor);
    }
}

File: solutions/problem-003/solution.cpp
#include <bits/stdc++.h>
using namespace std;

long long largest_prime_factor(long long n)
{
    long long i = 2LL;

    while (i * i <= n)
    {
        while (n % i == 0)
            n /= i;

        i++;
    }

    return n;
}

int main()
{

    long long n = 600851475143;
    cout << "Largest Prime Factor is : " << largest_prime_factor(n);

    return 0;
}
